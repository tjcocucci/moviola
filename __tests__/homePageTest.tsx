import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import 'react-datepicker/dist/react-datepicker.css';
import Home from '../src/app/page';

describe('[ app / Home ]', () => {
  describe('when component is rendered', () => {
    it('should render a the snapshoted home component', () => {
      const { asFragment } = render(<Home />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
