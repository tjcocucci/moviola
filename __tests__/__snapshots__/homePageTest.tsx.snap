// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[ app / Home ] > when component is rendered > should render a the snapshoted home component 1`] = `
<DocumentFragment>
  <div>
    <div
      class="p-4 flex self-center flex-auto flex-col gap-5 max-w-4xl"
    >
      <div
        class="mb-3"
      >
        <input
          class="p-2 border border-gray-300 rounded mr-2"
          placeholder="Search..."
          type="text"
          value=""
        />
      </div>
      <div
        class="flex"
      >
        <div
          class="flex items-center"
        >
          <span
            class="whitespace-nowrap mr-5"
          >
            Start date:
          </span>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container "
            >
              <span
                aria-live="polite"
                class="react-datepicker__aria-live"
                role="alert"
              />
              <input
                class="p-2 border border-gray-300 rounded w-2/3"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="flex items-center"
        >
          <span
            class="whitespace-nowrap mr-5"
          >
            End date:
          </span>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container "
            >
              <span
                aria-live="polite"
                class="react-datepicker__aria-live"
                role="alert"
              />
              <input
                class="p-2 border border-gray-300 rounded w-2/3"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="flex items-center "
        >
          <div
            class="p-2 "
          >
            <select>
              <option
                value=""
              >
                Order by
              </option>
              <option
                value="popularity.desc"
              >
                Popularity
              </option>
              <option
                value="release_date.desc"
              >
                Release date
              </option>
            </select>
          </div>
        </div>
      </div>
    </div>
    <div
      class="w-full h-px bg-white"
    />
    <div
      class="text-3xl my-5 font-bold mb-8 text-gray-800 self-center italic text-blue-700"
      style="text-shadow: 2px 2px 4px rgba(255, 255, 255, 0.5), -2px -2px 4px rgba(0, 0, 0, 0.5);"
    >
      Loading...
    </div>
    <div
      class="container my-10 mx-auto  max-w-4xl"
    >
      <div
        class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 gap-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
